File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/adme/AdmeGuideGrooveCalibrationEntity.java
package com.shmedo.configlibrary.iot.cmd.entity.adme;

import android.text.TextUtils;

import com.shmedo.configlibrary.ble.interfaces.Validater;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  1/8/21 <br/>
 * 描述：      生成ADME导槽校准配置参数拼接指令
 */
public class AdmeGuideGrooveCalibrationEntity  implements Validater {
    private String movementway;//运动方式（0:正转，1:反转）
    private String motorspeed;//电机速度
    private String movepulse;//运动脉冲数

    public void setMovementway(String movementway) {
        this.movementway = movementway;
    }

    public void setMotorspeed(String motorspeed) {
        this.motorspeed = motorspeed;
    }

    public void setMovepulse(String movepulse) {
        this.movepulse = movepulse;
    }

    @Override
    public void validate() {

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("movementway=" + movementway);
        stringBuilder.append("&");
        if (!TextUtils.isEmpty(motorspeed)) {
            stringBuilder.append("motorspeed=" + motorspeed);
            stringBuilder.append("&");
        }
        if (!TextUtils.isEmpty(movepulse)) {
            stringBuilder.append("movepulse=" + movepulse);
            stringBuilder.append("&");
        }

        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length() - 1, stringBuilder.length());
        }

        return stringBuilder.toString();
    }
}

File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/adme/AdmeGuideGrooveCalibrationEntity.kt
package com.shmedo.configlibrary.iot.cmd.entity.adme

import com.shmedo.configlibrary.ble.interfaces.Validater

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  1/8/21 <br></br>
 * 描述：      生成ADME导槽校准配置参数拼接指令
 */
class AdmeGuideGrooveCalibrationEntity : Validater {
    var movementway //运动方式（0:正转，1:反转）
            : String? = null
    var motorspeed //电机速度
            : String? = null
    var movepulse //运动脉冲数
            : String? = null

    override fun validate() {}
    override fun toString(): String {
        val stringBuilder = StringBuilder()
        try {
            for (f in javaClass.declaredFields) {
                val value = f[this]
                if (value != null && value != "NullKey") {
                    stringBuilder.append(f.name)
                    stringBuilder.append("=")
                    stringBuilder.append(value)
                    stringBuilder.append("&")
                }
            }
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length - 1, stringBuilder.length)
        }
        return stringBuilder.toString()
    }
}